src/MongoDB.Driver/Linq/Linq3Implementation/Serializers/KnownSerializers/KnownSerializersNode.cs

```diff
@@ -82,7 +85,7 @@ private HashSet<IBsonSerializer> GetPossibleSerializersAtThisLevel(Type type)
             var possibleSerializers = new HashSet<IBsonSerializer>();
             foreach (var serializer in _knownSerializers.Values.SelectMany(hashset => hashset))
             {
-                if (serializer.ValueType.IsAssignableFrom(type))
+                if (serializer.ValueType == type || serializer.ValueType.IsEnum() && type.IsNumericUnderlyingEnumType())
```

Comment:
I think we can get rid of the `IsNumericUnderlyingEnumType` method entirely by using this instead:

    var valueType = serializer.ValueType;
    if (valueType == type || valueType.IsEnum() && Enum.GetUnderlyingType(valueType) == type)